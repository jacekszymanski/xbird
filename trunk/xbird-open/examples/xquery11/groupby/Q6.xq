<result>{
  for $sales in doc("sales-records.xml")/*/record
  let $storeno := $sales/store-number,
    $product := doc("products.xml")/*/product[name = $sales/product-name],
    $prd := $product,
    $profit := $sales/qty * ($prd/price - $prd/cost)
  group by $storeno
  let $total-store-profit := sum($profit)
  where $total-store-profit > 100
  order by $total-store-profit descending
  return
    <store number = "{$storeno}" total-profit = "{$total-store-profit}"/> 
 }</result>
