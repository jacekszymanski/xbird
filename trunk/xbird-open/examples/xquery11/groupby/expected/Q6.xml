<result>
  <store number="3" total-profit="7320"/>
  <store number="2" total-profit="3030"/>
  <store number="1" total-profit="2100"/>
</result>
